api/tests/providers/allocine_movie_list_test.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+
+from pcapi.connectors.api_allocine import ALLOCINE_API_URL
+from pcapi.core.offers.models import Product
+from pcapi.core.providers.allocine_movie_list import synchronize_products
+
+from tests.domain import fixtures
+
+
+@pytest.mark.usefixtures("db_session")
+class AllocineMovieListTest:
+    def _configure_api_responses(self, requests_mock):
+        requests_mock.get(f"{ALLOCINE_API_URL}/movieList?after=", json=fixtures.ALLOCINE_MOVIE_LIST_PAGES[""])
+        requests_mock.get(
+            f"{ALLOCINE_API_URL}/movieList?after=YXJyYXljb25uZWN0aW9uOjQ5",
+            json=fixtures.ALLOCINE_MOVIE_LIST_PAGES["YXJyYXljb25uZWN0aW9uOjQ5"],
+        )
+
+    def test_synchronize_products_creates_products(self, requests_mock):
+        # Given
+        self._configure_api_responses(requests_mock)
+        assert Product.query.count() == 0
+
+        # When
+        synchronize_products()
+
+        # Then
+        catalogue = Product.query.order_by(Product.id).all()
+        assert len(catalogue) == 4
+
+        movie_data = catalogue[0].extraData
+        expected_data = fixtures.ALLOCINE_MOVIE_LIST_PAGES[""]["movieList"]["edges"][0]["node"]
+        assert movie_data["allocineId"] == expected_data["internalId"]
+        assert movie_data["backlink"] == expected_data["backlink"]["url"]
+        assert movie_data["cast"] == [
+            f"{item['node']['actor']['firstName']} {item['node']['actor']['lastName']}"
+            for item in expected_data["cast"]["edges"]
+        ]
+        assert movie_data["companies"] == [
+            {"activity": company["activity"], "name": company["company"]["name"]}
+            for company in expected_data["companies"]
+        ]
+        assert movie_data["countries"] == [country["name"] for country in expected_data["countries"]]
+        assert movie_data["credits"] == [expected_data["credits"]["edges"][0]["node"]]
+        assert movie_data["data"] == expected_data["data"]
+        assert not movie_data.get("diffusionVersion")
```

Comment:
mineur, mais je comprends de cette ligne que `movie_data["diffusionVersion"]` est falsy, si tu veux être plus spécifique tu peux écrire `"diffusionVersion not in movie_data`

idem to `theater` plus bas